pkg/kubelet/kubelet.go

```diff
@@ -1065,6 +1065,46 @@ func (kl *Kubelet) cleanupOrphanedVolumes(pods []*api.Pod, runningPods []*kubeco
 	return nil
 }
 
+// filterOutPodsExceedRuntimeTtl filters pods that have a container whose startAt time is more
+// than the specified RunningTtl for the pod.  It records an event that the pod has run longer
+// than the allocated time, and updates the pod status as failed.  By filtering the pod from
+// the result set, the Kubelet will kill the pod's containers as part of normal SyncPods workflow
```

Comment:
Goal is to kill pod and mark failed so it's independent of restart policy.  Issue is getting a good stable date and time to make the relative value.  It looked like taking the oldest container started at was insufficient because it did change if there ever was a container restart.  So in my local branch I tried to keep a StartTime aside from the container statuses in status manager.  Will update for review after I push an updated branch.